engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1776,19 +1773,26 @@ private void orchestrateStorageMigration(final String vmUuid, final StoragePool
                 // If VM was cold migrated between clusters belonging to two different VMware DCs,
                 // unregister the VM from the source host and cleanup the associated VM files.
                 if (vm.getHypervisorType().equals(HypervisorType.VMware)) {
+                    Long srcClusterId = null;
+                    Long srcHostId = vm.getHostId() != null ? vm.getHostId() : vm.getLastHostId();
+                    if (srcHostId != null) {
+                        HostVO srcHost = _hostDao.findById(srcHostId);
```

Comment:
@alexandrelimassantana I don't see any problem with that **if** (line 1778). Both variables (**srcHostId** and **srcClusterId**) are used.

The conditional at the line 1784 uses **srcClusterId**.

The method at line 1793 sends the **UnregisterVMCommand** to a given host with the **srcHostId**.